### ISSUE TITLE
isAllowedToRedeemBy cannot check the owner

### ISSUE THREAD

**Issue by winnie-hypo** [michael-rubel/laravel-couponables#23]
my coupon:
redeemer_type: App\Models\Member,
redeemer_id: 30,

$redeemer = Member::find(18);
$redeemer->verifyCoupon($coupon_code); //return no exception
 
no matter which redeemer it is, always pass the validation.

i found the issue in this function.

```
public function isAllowedToRedeemBy(Model $redeemer): bool
    {
        return with(static::$bindable, function ($coupon) use ($redeemer) {
            if ($coupon->isMorphColumnsFilled() && ! $coupon->redeemer?->is($redeemer)) {
                return false;
            }

            if ($coupon->isOnlyRedeemerTypeFilled() && ! $coupon->isSameRedeemerModel($redeemer)) {
                return false;
            }

            return true;
        });
    }

```

$coupon is always null inside this function >> $coupon->redeemer_type and $coupon->redeemer_id are null >> always return true.

and if i use $this instead of static::$bindable, everything works.

Please help to fix this issue, thanks!

**Comment by michael-rubel** [comment]
Hello @winnie-hypo 

Just did the same in this test: #24 
And it seems to be working correctly.

By any chance, your model bypasses the trait initialization?
Please, debug the `initializeHasCoupons` method and if it works, then check the `CallProxy` class.
Also, specify the version of [`LEC`](https://github.com/michael-rubel/laravel-enhanced-container) your app uses.

**Comment by winnie-hypo** [comment]
/** @test */
    public function testCanVerifyCouponWhenSpecificModelAssigned()
    {
        Coupon::where('code', 'redeemer-coupon')->delete();

        Coupon::create([
            'code'          => 'redeemer-coupon',
            'redeemer_type' => Member::class,
            'redeemer_id'   => 30,
            'limit' => 1,
            'quantity'=>1,
        ]);

        $redeemer = Member::find(18);

        $redeemed = $redeemer->verifyCoupon('redeemer-coupon');

        $this->assertInstanceOf(Coupon::class, $redeemed);

    }



i did this test, if i set limit = 1, other redeemer can redeem the coupon. please help me, thxx

**Comment by michael-rubel** [comment]
Caught and fixed it. :)
Thanks for reporting!

Fix released in [v7.1.2](https://github.com/michael-rubel/laravel-couponables/releases/tag/7.1.2)

**Comment by winnie-hypo** [comment]
THANKS
